### PR TITLE
Migrate airflow-database guide

### PR DESCRIPTION
**Pull request by kwcanuck**
Migration of the existing guide airflow-database (Understanding the Airflow Metadata Database)

### CODE REVIEW

File: guides/airflow-database.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+> **Note**: For many use cases you can access contents from the metadata database in the Airflow UI or the [stable REST API](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html). These points of access always beat querying the metadata database directly!
+
+There are several types of metadata stored in the metadata database.
+
+- User login information and permissions.
+- Information used in DAGs, like variables, connections and XComs.
+- Data about DAG and task runs which are generated by the scheduler.
+- Other minor tables, such as tables which store DAG code in different formats or information about import errors.
+
+Most of this data can be queried using the [Airflow REST API](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html).
```
Comment (jwitz) [inline guides/airflow-database.md:52]:
```suggestion
For many use cases you can access content from the metadata database in the Airflow UI or the [stable REST API](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html). These points of access always beat querying the metadata database directly!
```

File: guides/airflow-database.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+A healthy metadata database is critical for your Airflow environment. Losing data stored in the metadata database can both interfere with running DAGs and prevent you from accessing data for past DAG runs. As with any core Airflow component, having a backup and disaster recovery plan in place for the metadata database is essential.
+
+In this guide, you'll learn everything you need to know about the Airflow metadata database to ensure a healthy Airflow environment, including:
+
+- Database specifications.
+- Important content stored in the database.
+- Best practices for using the metadata database.
+- Different methods for accessing data of interest.
+
+> **Note**: Astronomer strongly advise against directly modifying the metadata database, as this can cause dependency issues and corrupt your Airflow instance.
```
Comment (jwitz) [inline guides/airflow-database.md:20]:
```suggestion
```

File: guides/airflow-database.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+# from the API response print the value for "total entries"
+print(req.json()['total_entries'])
+```
+
+It is also possible to navigate to **Browse** -> **Task Instances** in the Airflow UI and filter the task instances for all with a state of `success`. The `Record Count` will be on the right side of your screen.
+
+![Count successful tasks Airflow UI](/img/guides/successful_tasks_UI.png)
+
+### Example: Pause and unpause a DAG
```
Comment (jwitz) [inline guides/airflow-database.md:141]:
```suggestion
### Pause and unpause a DAG
```

File: guides/airflow-database.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+# query the API to patch all tasks as paused
+req = requests.patch(
+  f"{ENDPOINT_URL}/api/v1/dags/{dag_id}?update_mask=is_paused", json=update,
+  auth=(user_name, password))
+
+# print the API response
+print(req.text)
+```
+
+### Example: Delete a DAG
```
Comment (jwitz) [inline guides/airflow-database.md:174]:
```suggestion
### Delete a DAG
```

File: guides/airflow-database.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+# send the deletion request
+req = requests.delete(
+  f"{ENDPOINT_URL}/api/v1/dags/{dag_id}",
+  auth=(user_name, password))
+
+# print the API response
+print(req.text)
+```
+
+### Example: Retrieve all DAG dependencies
```
Comment (jwitz) [inline guides/airflow-database.md:205]:
```suggestion
### Retrieve all DAG dependencies
```

File: guides/airflow-database.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+conn_url = f'{sql_alchemy_conn}/postgres'
+
+engine = create_engine(conn_url)
+
+with Session(engine) as session:
+    result = session.query(SerializedDagModel).first()
+    print(result.get_dag_dependencies())
+```
+
+### Example: Retrieve Alembic version
```
Comment (jwitz) [inline guides/airflow-database.md:230]:
```suggestion
### Retrieve alembic version
```

File: guides/airflow-database.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
+---
+title: "Understanding the Airflow metadata database"
+sidebar_label: "Understanding the Airflow metadata database"
```
Comment (jwitz) [inline guides/airflow-database.md:3]:
```suggestion
sidebar_label: "Metadata database"
```

File: guides/airflow-database.md [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+description: "A walkthrough of the Apache Airflow metadata database, with a full ERD."
+id: airflow-database
+tags: ["Database", "SQL", "Components"]
```
Comment (jwitz) [inline guides/airflow-database.md:6]:
```suggestion
tags: [Database, SQL, Components]
```

File: guides/airflow-database.md [hunk 9]
```diff
@@ -0,0 +1,256 @@
+---
+title: "Understanding the Airflow metadata database"
+sidebar_label: "Understanding the Airflow metadata database"
+description: "A walkthrough of the Apache Airflow metadata database, with a full ERD."
```
Comment (jwitz) [inline guides/airflow-database.md:4]:
```suggestion
description: "Learn about everything you need to use the Apache Airflow metadata database."
```

**Review by jwitz — APPROVED** [review]
Few more copy edits in this one, mainly related to our use of notes

File: guides/airflow-database.md [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+
+There are additional tables in the metadata database storing data ranging from DAG tags over serialized DAG code, import errors to current states of sensors. Some of the information in these tables will be visible in the Airflow UI in various places:
+
+- The source code of DAGs can be found by clicking on a DAG name from the main view and then going to the **Code** view.
+- Import errors appear at the top of the **DAGs** view in the UI.
+- DAG tags will appear underneath their respective DAG with a cyan background.
+
+## Airflow metadata database best practices
+
+1. When upgrading or downgrading Airflow, always follow the [recommended steps for changing Airflow versions](https://airflow.apache.org/docs/apache-airflow/stable/installation/upgrading.html?highlight=upgrade): back up the metadata database, check for deprecated features, pause all DAGs, and make sure no tasks are running.
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply guides/airflow-database.md]: @jwitz - done!

**Pull request merged**